esp-hal/src/soc/esp32s3/cpu_control.rs

```diff
@@ -227,6 +228,13 @@ impl<'d> CpuControl<'d> {
         unsafe {
             xtensa_lx::set_vecbase(&raw const _init_start);
             xtensa_lx::set_stack_pointer(APP_CORE_STACK_TOP.load(Ordering::Acquire));
+
+            #[cfg(all(feature = "rt", stack_guard_monitoring))]
+            {
+                let stack_guard = APP_CORE_STACK_GUARD.load(Ordering::Acquire);
+                // setting 0 effectively disables the functionality
+                crate::debugger::set_stack_watchpoint(stack_guard as usize);
```

Comment:
The stack guard value should be written before enabling the watchpoint on the second core

Reply:
we don't write the stack guard yet - but why not might help if someone is manually inspecting the stack